packages/ui/src/council/modals/RecoverVoteStake/RecoverVoteStakeModal.tsx

```diff
@@ -0,0 +1,53 @@
+import { useMachine } from '@xstate/react'
+import React, { useEffect, useMemo } from 'react'
+
+import { useTransactionFee } from '@/accounts/hooks/useTransactionFee'
+import { InsufficientFundsModal } from '@/accounts/modals/InsufficientFundsModal'
+import { FailureModal } from '@/common/components/FailureModal'
+import { useApi } from '@/common/hooks/useApi'
+import { useModal } from '@/common/hooks/useModal'
+import { isDefined } from '@/common/utils'
+
+import { RecoverVoteStakeModalCall } from '.'
+import { recoverVoteStakeMachine } from './machine'
+import { RecoverVoteStakeSignModal } from './RecoverVoteStakeSignModal'
+import { RecoverVoteStakeSuccessModal } from './RecoverVoteStakeSuccessModal'
+
+export const RecoverVoteStakeModal = () => {
+  const [state, send] = useMachine(recoverVoteStakeMachine)
+  const { api } = useApi()
+  const {
+    hideModal,
+    modalData: { address },
+  } = useModal<RecoverVoteStakeModalCall>()
+  const transaction = useMemo(() => api?.tx.referendum.releaseVoteStake(), [])
```

Comment:
Probably would not fail, but the deps of `useMemo()` will cause to render the hook only once.

It should be: `[api]`, because it is using the `api` inside memo.

Reply:
see: https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies
 
If you specify a list of dependencies as the last argument to useEffect, useLayoutEffect, useMemo, useCallback, or useImperativeHandle, it must include all values that are used inside the callback and participate in the React data flow. That includes props, state, and anything derived from them